kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -118,35 +115,33 @@ impl<'a> UnitAnalyzer<'a> {
     /// generate the OffsetIndex
     pub fn handle_files(&mut self) -> Result<(), KytheError> {
         // https://kythe.io/docs/schema/#file
-        for source_file in self.unit.get_source_file() {
-            let vname = self.get_file_vname(source_file)?;
-
+        for (file_path, vname) in &self.file_vnames {
```

Comment:
For languages-not-C++ the indexer should generally restrict itself to indexing the files listed in the source_file attribute rather than all of the required_input.

Do you have an example of where the path from `source_file` isn't present in `required_input`?  It's likely that the extractor should be normalizing and relativizing paths.

Reply:
If a generated file is used as compiler input, the path after passing through the vname rules (in required_input) is different than the path that is present in path used by the compiler and in the save_analysis file.

Reply:
Currently, only rust source files are added to `required_input` in the extractor, so `source_file` === `required_input`, just with different paths. Whether the extractor should be changed to add dependency libraries may be an additional conversation to be had.

Reply:
Right, that's as expected.  There are two paths in required_input: `required_input.info.path` and `required_input.v_name.path`. The former should be used for finding compiler paths and should match the `source_file` (and save_analysis) path for generated files.  The latter should rarely be used directly as it is largely meaningless without the corresponding root and corpus.

Does the Rust extractor not include information about dependencies in `required_input`? Using a hypothetical BUILD rules for an example:

```
rust_binary(name = "bin", srcs = ["main.rs"], deps = [":lib"])
rust_library(name = "lib", srcs = ["lib.rs"])
```

The compilation unit for `bin` should only include `main.rs` in the `source_file` but would include whatever files were necessary to reference entities arising from `lib.rs` in `required_input`.

Reply:
Ah I see what you are saying regarding `required_input.info.path`, I forgot that field existed. No information about dependencies are kept in `required_input`, as it would require saving compiled binary `.rlib` files to the kzip. The extractor knows nothing about the source files that are used to build the dependencies.

Reply:
If you have file paths referenced from the save_analysis file which have no entries in `required_input`, the extractor needs to put that information somewhere. I think a language-specific message in the `details` field that is a simple `map<string, VName>` would suffice.  It could also be limited to the cases where `(corpus, root, path)` differ from the simple case.